lib/pf/OMAPI.pm

```diff
@@ -0,0 +1,547 @@
+package pf::OMAPI;
+=head1 NAME
+
+pf::OMAPI
+
+=cut
+
+=head1 DESCRIPTION
+
+pf::OMAPI
+
+=head1 SYNOPSIS
+
+Quick summary of what the module does.
+
+Perhaps a little code snippet.
+
+    use pf::OMAPI;
+
+    my $omapi = pf::OMAPI->new( key_name => 'pf_omapi_key',key_base64 => 'xJviCHiQKcDu6hk7+Ffa3A==', host => 'localhost', port => 7911);
+
+    my $data = $omapi->lookup({'ip-address' => "10.229.25.247" });
+
+=cut
+
+use strict;
+use warnings;
+use Moo;
+use MIME::Base64;
+use Net::IP;
+use Digest::HMAC_MD5 qw(hmac_md5);
+use IO::Socket::INET;
+use Socket qw(MSG_WAITALL);
+
+our $VERSION = '0.01';
+
+
+=head1 ATTRIBUTES
+
+=head2 host
+
+host of dhcp server
+
+=cut
+
+has host => (is => 'rw', default => sub  { 'localhost' });
+
+=head2 port
+
+port of the dhcp server
+
+=cut
+
+has port => (is => 'rw', default => sub { 7911 } );
+
+=head2 buffer
+
+The reference to the buffer of message
+
+=cut
+
+has buffer => (is => 'rw', default => sub { my $s = "";\$s } );
+
+=head2 sock
+
+The socket used for communicating with the dhcpd server
+
+=cut
+
+has sock => (is => 'rw', builder => 1, lazy => 1, clearer => 1);
+
+=head2 connected
+
+A check if we are connected to the omapi service
+
+=cut
+
+has connected => (is => 'rw' , default => sub { 0 } );
+
+=head2 key_name
+
+The name of the key
+
+=cut
+
+has key_name => (is => 'rw');
+
+=head2 op
+
+The current operation
+
+=cut
+
+has op => (is => 'rw');
+
+=head2 msg
+
+The current message to be sent
+
+=cut
+
+has msg => (is => 'rw');
+
+=head2 obj
+
+The current obj to be sent
+
+=cut
+
+has obj => (is => 'rw');
+
+=head2 authid
+
+The auth id to send
+
+=cut
+
+has authid => (is => 'rw', default => sub { 0 });
+
+
+=head2 authlen
+
+The length of signature
+
+=cut
+
+has authlen => (is => 'rw', default => sub { 0 });
+
+=head2 id
+
+The current id of the message
+
+=cut
+
+has id => (is => 'rw', default=> sub { int(rand(0x10000000)) } );
+
+=head2 key
+
+The key used to sign messages
+
+=cut
+
+has key => (is => 'rw', builder => 1, lazy => 1);
+
+=head2 key
+
+The key base64 representation of the key
+
+=cut
+
+has key_base64 => (is => 'rw');
+
+#The different message types
+
+our $OPEN    = 1;
+our $REFRESH = 2;
+our $UPDATE  = 3;
+our $NOTIFY  = 4;
+our $ERROR   = 5;
+our $DELETE  = 6;
+
+
+#The unpack format for the different return dhcp objects
+#
+
+our %FORMATLIST = (
+    'flags'         => 'C',
+    'ends'          => 'N',
+    'tstp'          => 'N',
+    'tsfp'          => 'N',
+    'cltt'          => 'N',
+    'pool'          => 'N',
+    'state'         => 'N',
+    'atsfp'         => 'N',
+    'starts'        => 'N',
+    'subnet'        => 'N',
+    'hardware-type' => 'N',
+    'result'        => 'N',
+);
+
+
+#The methods used to unpack special values
```

Comment:
This is internal to the module.
%UNPACK_DATA will not be used outside of the module.